Swen/Classes/Swen.swift

```diff
@@ -86,18 +118,25 @@ fileprivate extension Swen {
         listeners.append(listener)
         if let sticky = sticky {
             listener.queue.addOperation {
-                listener.post(sticky)
+                handler(sticky)
```

Comment:
Why would there be a deadlock if it is the same thread?
Let's say both are on the main queue:
Register->Listener.init()->listener.post()->handler()

Where would the deadlock occur?